bitcoin/src/address/mod.rs

```diff
@@ -376,16 +378,19 @@ impl Address {
     /// This address type was introduced with BIP16 and is the popular type to implement multi-sig
     /// these days.
     #[inline]
-    pub fn p2sh(script: &Script, network: impl Into<NetworkKind>) -> Result<Address, P2shError> {
-        if script.len() > MAX_SCRIPT_ELEMENT_SIZE {
-            return Err(P2shError::ExcessiveScriptSize);
-        }
-        let hash = script.script_hash();
+    pub fn p2sh(
+        redeem_script: &Script,
+        network: impl Into<NetworkKind>,
+    ) -> Result<Address, RedeemScriptSizeError> {
+        let hash = redeem_script.script_hash()?;
         Ok(Address::p2sh_from_hash(hash, network))
     }
 
-    // This is intentionally not public so we enforce script length checks.
-    fn p2sh_from_hash(hash: ScriptHash, network: impl Into<NetworkKind>) -> Address {
+    /// Creates a pay to script hash P2SH address from a script hash.
+    ///
+    /// **Warning** The `hash` pre-image (redeem script) must not exceed 520 bytes in length
```

Comment:
I think that we are using something like this from the standardization in #2798:

```suggestion
    /// # Warning
    ///
    /// The `hash` pre-image (redeem script) must not exceed 520 bytes in length
```

Reply:
Fixed in #2795, thanks.